debacl/level_set_tree.py

```diff
@@ -1260,6 +1260,72 @@ def construct_tree(X, k, prune_threshold=None, num_levels=None, verbose=False):
     return tree
 
 
+def construct_tree_from_precomputed_matrix(X, p, k, prune_threshold=None, num_levels=None, verbose=False):
```

Comment:
Can we change "precomputed_matrix" to "distance_matrix"? I think it would be a little more explicit.